packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -1129,6 +1120,22 @@ class FlutterPlugin implements Plugin<Project> {
             ) {
                 dependsOn compileTask
                 with compileTask.assets
+                def currentGradleVersion = project.getGradle().getGradleVersion()
+
+                // See https://docs.gradle.org/current/javadoc/org/gradle/api/file/ConfigurableFilePermissions.html
+                // See https://github.com/flutter/flutter/pull/50047
+                if (compareVersionStrings(currentGradleVersion, '8.3') >= 0 ) {
+                    filePermissions {
+                        user {
+                            read = true
+                            write = true
+                        }
+                    }
+                } else {
+                    // See https://docs.gradle.org/8.2/dsl/org.gradle.api.tasks.Copy.html#org.gradle.api.tasks.Copy:fileMode
+                    // See https://github.com/flutter/flutter/pull/50047
+                    fileMode 0644
```

Comment:
this works from windows?

Reply:
I believe so but dont have a windows machine to check. The test passes on windows, this is the gradle api for setting file mode which works on windows. There is not a windows specific api.

Reply:
acknowledged. my understanding is windows' concept of permissions is much less granular than unix and thus there must be some primitive mapping of the octal codes to windows land inside gradle...

...but if the test passes that seems good enough :)